R/arc-select.R

```diff
@@ -264,6 +258,67 @@ collect_layer <- function(
   res
 }
 
+#' Get query responses with handling for layers that don't support pagination
+#' @noRd
+get_query_resps <- function(req,
+                            x,
+                            n_feats,
+                            page_size = NULL,
+                            query_params = list(),
+                            error_call = caller_env()) {
+  if (isFALSE(x[["advancedQueryCapabilities"]][["supportsPagination"]])) {
+    if (n_feats > x[["maxRecordCount"]]) {
+      cli::cli_abort(
+        "query can't be completed if {class(x)} does not support pagination and
+        the number of selected features exceeds the maximum record count.",
+        call = error_call
+      )
+    }
+
+    req <- httr2::req_url_path_append(req, "query")
+
+    req_query <- rlang::exec(
+      httr2::req_url_query,
+      .req = req,
+      !!!query_params
+    )
+
+    # If url is more than 2048 characters long, add the query to the
+    # body of the request
+    if (nchar(req_query[["url"]]) > 2048) {
+      req_query <- rlang::exec(
+        httr2::req_body_form,
+        .req = req,
+        !!!query_params
+      )
+    }
```

Comment:
I think we should always prefer using form body here. is there any reason to prefer a get request here?